accounts/abi/event_test.go

```diff
@@ -54,3 +85,80 @@ func TestEventId(t *testing.T) {
 		}
 	}
 }
+
+func TestEventTupleUnpack(t *testing.T) {
+	type EventTransfer struct {
+		Value *big.Int
+	}
+
+	type EventPledge struct {
+		Who      common.Address
+		Wad      *big.Int
+		Currency [3]byte
+	}
+
+	var testCases = []struct {
+		data     string
+		dest     interface{}
+		expected interface{}
+		jsonLog  []byte
+		name     string
+	}{{
+		"00000000000000000000000000000000000000000000000000000000000f4240",
+		&EventTransfer{},
+		&EventTransfer{Value: big.NewInt(1000000)},
+		jsonEventTransfer,
+		"Can unpack ERC20 Transfer event into structure",
+	}, {
+		"00000000000000000000000000000000000000000000000000000000000f4240",
+		&[]interface{}{new(big.Int)},
+		&[]interface{}{big.NewInt(1000000)},
+		jsonEventTransfer,
+		"Can unpack ERC20 Transfer event into slice",
+	}, {
+		"00000000000000000000000000ce0d46d924cc8437c806721496599fc3ffa2680000000000000000000000000000000000000000000000000000020489e800007573640000000000000000000000000000000000000000000000000000000000",
+		&EventPledge{},
+		&EventPledge{
+			common.HexToAddress("0x00Ce0d46d924CC8437c806721496599FC3FFA268"),
+			big.NewInt(2218516807680),
+			[3]byte{'u', 's', 'd'}},
+		jsonEventPledge,
+		"Can unpack Pledge event into structure",
+	}, {
+		"00000000000000000000000000ce0d46d924cc8437c806721496599fc3ffa2680000000000000000000000000000000000000000000000000000020489e800007573640000000000000000000000000000000000000000000000000000000000",
+		&[]interface{}{common.Address{}, new(big.Int), [3]byte{}},
+		&[]interface{}{
+			common.HexToAddress("0x00Ce0d46d924CC8437c806721496599FC3FFA268"),
+			big.NewInt(2218516807680),
+			[3]byte{'u', 's', 'd'}},
+		jsonEventPledge,
+		"Can unpack Pledge event into slice",
+	}}
```

Comment:
All the `Slice` based tests were failing because of the `Elem()` call in `tupleUnpack`:

- Can unpack ERC20 Transfer event into slice
- Can unpack Pledge event into slice